tests/perfalert/test_alerts/test_secretary_tool.py

```diff
@@ -89,3 +184,103 @@ def test_secretary_tool_creates_new_settings_if_none_exist(db):
     SecretaryTool.validate_settings()
 
     assert PerformanceSettings.objects.count() == 1
+
+
+def test_check_outcome_after_success(get_outcome_checker_mock, record_backfilled):
+    outcome_checker_mock = get_outcome_checker_mock(OutcomeStatus.SUCCESSFUL)
+    secretary = SecretaryTool(outcome_checker_mock)
+
+    assert BackfillRecord.objects.filter(status=BackfillRecord.BACKFILLED).count() == 1
+    assert BackfillRecord.objects.filter(status=BackfillRecord.SUCCESSFUL).count() == 0
+    secretary.check_outcome()
+    assert BackfillRecord.objects.filter(status=BackfillRecord.BACKFILLED).count() == 0
+    assert BackfillRecord.objects.filter(status=BackfillRecord.SUCCESSFUL).count() == 1
+
+
+def test_check_outcome_after_fail(get_outcome_checker_mock, record_backfilled):
+    outcome_checker_mock = get_outcome_checker_mock(OutcomeStatus.FAILED)
+    secretary = SecretaryTool(outcome_checker_mock)
+
+    assert BackfillRecord.objects.filter(status=BackfillRecord.BACKFILLED).count() == 1
+    assert BackfillRecord.objects.filter(status=BackfillRecord.FAILED).count() == 0
+    secretary.check_outcome()
+    assert BackfillRecord.objects.filter(status=BackfillRecord.BACKFILLED).count() == 0
+    assert BackfillRecord.objects.filter(status=BackfillRecord.FAILED).count() == 1
+
+
+def test_no_action_when_in_progress(get_outcome_checker_mock, record_backfilled):
+    outcome_checker_mock = get_outcome_checker_mock(OutcomeStatus.IN_PROGRESS)
+    secretary = SecretaryTool(outcome_checker_mock)
+
+    assert BackfillRecord.objects.filter(status=BackfillRecord.BACKFILLED).count() == 1
+    secretary.check_outcome()
+    assert BackfillRecord.objects.filter(status=BackfillRecord.BACKFILLED).count() == 1
+
+
+def test_outcome_checker_identifies_pushes_in_range(
+    record_backfilled, test_repository_2, range_dates, outcome_checking_pushes
+):
+    outcome_checker = OutcomeChecker()
+    total_pushes = Push.objects.count()
+
+    from_time = range_dates['from_date']
+    to_time = range_dates['to_date']
+    repository_id = 1
+
+    outside_range = Push.objects.filter(
+        Q(repository_id=repository_id) & (Q(time__lt=from_time) | Q(time__gt=to_time))
+    ).count()
+
+    pushes_in_range = outcome_checker._get_pushes_in_range(from_time, to_time, repository_id)
+    assert len(pushes_in_range) == total_pushes - outside_range
+
+    # change repository for the first 2 pushes in range
+    assert repository_id != test_repository_2.id
+
+    number_of_changed_pushes = 2
```

Comment:
This is a big name. What do you think about replacing `number_of_` with `total_` prefix. Maybe even use it for all count-like variables.